### PR TITLE
Feature/add editorconfig

### PR DESCRIPTION
**Pull request by alexjoverm**

### CODE REVIEW

**Comment by alexjoverm** [comment]
@kentcdodds done!

**Comment by kentcdodds** [comment]
Same thing on this pull request as your other. Try an interactive rebase.

**Comment by alexjoverm** [comment]
I did, but since I pressed the auto-sync button of vscode I think it pulled and merge under the hood, instead of updating the upstream. I'll give it another try

**Comment by alexjoverm** [comment]
Done here as well!

**Comment by alexjoverm** [comment]
Well, some of them they really do, exactly:

```
end_of_line = lf
charset = utf-8
trim_trailing_whitespace = true
indent_size = 2
```

When I opened the project with my editor, it converted every file to EOL CLRF, causing having 2000+ linter errors by `xo`. And by looking at the project, those ones apply.

About the rest, I think is a matter of preference, so we can discuss them, like `indent_style`, `max_line_length`, etc. My personal opinion I would probably remove the `[*.md]` and `[*.{json,yml}]` part, and change to `indent_style = true`.

What are your opinions on this @kentcdodds @jfmengels?

**Comment by alexjoverm** [comment]
Any news about this guys?

**Comment by kentcdodds** [comment]
> When I opened the project with my editor, it converted every file to EOL CLRF

We should probably just add `.gitattributes` to set the line endings. Not everyone will have an editor plugin that works with `.editorconfig`

I'm fine with these changes, but I'm going to let @jfmengels merge it if he wants to.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#root = true
+
+[*]
+indent_style = tab
```
Comment (kentcdodds) [inline .editorconfig]: This should probably be `space`.

**Comment by alexjoverm** [comment]
@kentcdodds @jfmengels Sorry, forgot about this :(

Fixed the requested changes. The `.gitattributes` sounds like a great idea, didn't know that option. They could be complementary I guess :)

**Comment by jfmengels** [comment]
Thanks a lot @alexjoverm! I love editorconfig, it's super useful :)

**Pull request merged**